### PR TITLE
Add SemVer2.0 Support

### PR DESCRIPTION
**Pull request by ryuyu**
Update NuGet.Core reference to 2.14 for server 2.0
Expand semver tests to include semver 2.0 type versions.

@emgarten @shishirx34 @maartenba @xavierdecoster

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -10,8 +10,11 @@
   <packageSources>
     <clear />
     <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
+    <add key="NuGet-Volatile" value="https://dotnet.myget.org/F/nuget-volatile" />
```
Comment (shishirx34) [inline NuGet.config]: is nuget.core 2.14 on nuget-volatile feed?

**Comment by emgarten** [comment]
Would you add tests to ``NormalizeVersionInterceptorTest`` and ``ServerPackageRepositoryTest`` also?

Are you planning to update this again with 2.14 RTM? Maybe 2.13 and 2.14 should just be moved to RTM now.

**Comment by ryuyu** [comment]
I'm just putting this out for now to get feedback. I'm not sure when 2.14 RTM will be available. If we are okay with it, i would be okay with RTMing 2.14 and updating the references to point to that release.

File: NuGet.config (hunk 1 above)
Comment (emgarten) [inline NuGet.config]: yes, and the volatile feed is used as an input for the nugetbuild feed.

File: test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs [hunk 2]
```diff
@@ -48,7 +51,7 @@ public void SerializesSemanticVersionAsString(string version)
                 converter.WriteJson(writer, SemanticVersion.Parse(version), new JsonSerializer());
 
                 // Assert
-                Assert.Equal("\"" + version + "\"", json.ToString());
+                Assert.Equal("\"" + expected + "\"", json.ToString());
```
Comment (shishirx34) [inline test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs]: wow.. I think luckily that all the cases for version and expected were the same this was never noticed.

File: test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs [hunk 3]
```diff
@@ -72,13 +75,21 @@ public void SerializesVersionAsString(string version)
...
+        [InlineData("4.0.0-0test.zero+tagParses", typeof(SemanticVersion), "4.0.0-0test.zero")]
+        [InlineData("4.0.0-test.more.parts+tagsHash", typeof(SemanticVersion), "4.0.0-test.more.parts")]
+        [InlineData("1.0.0", typeof(Version), "1.0.0")]
+        [InlineData("2.0.0.0", typeof(Version), "2.0.0.0")]
+        public void Deserializes(string version, Type type, string expected=null)
         {
+            if (expected == null)
+            {
+                expected = version;
+            }
```
Comment (shishirx34) [inline test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs]: don't think you need this anymore since you specify all `expected` cases above?

**Comment by shishirx34** [comment]
Looks good.

File: test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs (hunk 3 above)
Comment (ryuyu) [inline test/NuGet.Server.Core.Tests/SemanticVersionJsonConverterTests.cs]: Correct.

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**